### PR TITLE
CI: Test EC2 import and boot

### PR DESCRIPTION
**Pull request by major**
Build an AWS AMI image, upload it to S3, import it into EC2, and boot it.

Signed-off-by: Major Hayden <major@redhat.com>

### CODE REVIEW

**Comment by major** [comment]
😯

```
Logs from test case: fedora_32-x86_64-ami-boot.json
----------------------------------------------------------------------
stderr:
curl: (28) Connection timed out after 10000 milliseconds
----------------------------------------------------------------------
stdout:
=== RUN   TestImages
=== RUN   TestImages//usr/share/tests/osbuild-composer/cases/fedora_32-x86_64-ami-boot.json
=== PAUSE TestImages//usr/share/tests/osbuild-composer/cases/fedora_32-x86_64-ami-boot.json
=== CONT  TestImages//usr/share/tests/osbuild-composer/cases/fedora_32-x86_64-ami-boot.json
=== RUN   TestImages//usr/share/tests/osbuild-composer/cases/fedora_32-x86_64-ami-boot.json/image_info
=== RUN   TestImages//usr/share/tests/osbuild-composer/cases/fedora_32-x86_64-ami-boot.json/boot
--- PASS: TestImages (0.01s)
    --- PASS: TestImages//usr/share/tests/osbuild-composer/cases/fedora_32-x86_64-ami-boot.json (793.44s)
        --- PASS: TestImages//usr/share/tests/osbuild-composer/cases/fedora_32-x86_64-ami-boot.json/image_info (10.98s)
        --- PASS: TestImages//usr/share/tests/osbuild-composer/cases/fedora_32-x86_64-ami-boot.json/boot (585.50s)
PASS
```

**Comment by major** [comment]
@ondrejbudai Just cleaned up the PR and it should be okay to review now. Let's hope that the EC2 import works two times in a row. 🤞

**Comment by major** [comment]
Hmm, failed to register on the second try. I wonder if something broke in AWS during that test run.

**Comment by major** [comment]
#566 showed up again. Looking good otherwise.

**Comment by major** [comment]
@ondrejbudai Perhaps we need to tinker with the timeout after all. 🙃

```
cannot register the image: &awserr.baseError{code:"ResourceNotReady", message:"exceeded wait attempts", errs:[]error(nil)}
```

**Comment by major** [comment]
Not sure why the pull request test is angry at me. :/

**Comment by larskarlitski** [comment]
> Not sure why the pull request test is angry at me. :/

There's a whitespace error (yay golang):
```diff
 --- i/internal/upload/awsupload/awsupload.go
+++ w/internal/upload/awsupload/awsupload.go
@@ -73,7 +73,7 @@ func WaitUntilImportSnapshotTaskCompleted(c *ec2.EC2, input *ec2.DescribeImportS
 // for more information on using Contexts.
 func WaitUntilImportSnapshotTaskCompletedWithContext(c *ec2.EC2, ctx aws.Context, input *ec2.DescribeImportSnapshotTasksInput, opts ...request.WaiterOption) error {
 	w := request.Waiter{
-		Name:        "WaitUntilImportSnapshotTaskCompleted",
+		Name: "WaitUntilImportSnapshotTaskCompleted",
 		// NOTE(mhayden): Wait for 90 minutes for the snapshot to import.
 		// Some AWS regions are fairly busy with image imports and they can
 		// sometimes take a long time.
```

 You can fix it by running `./tools/prepare-source.sh` (or `go fmt ./...`).

**Comment by major** [comment]
Thanks, @larskarlitski. I'll see if I can figure out how that got in there.

**Comment by major** [comment]
AWS test completed in < 12 minutes! 🥳 🎉

However, the Fedora 32 weldr test [failed with a strange timeout](https://osbuildci.cloud.paas.psi.redhat.com/blue/organizations/jenkins/osbuild%2Fosbuild-composer/detail/PR-621/23/pipeline/18):
```
panic: test timed out after 5m0s
```

File: internal/upload/awsupload/awsupload.go [hunk 1]
```diff
@@ -71,10 +71,14 @@ func WaitUntilImportSnapshotTaskCompleted(c *ec2.EC2, input *ec2.DescribeImportS
...
 // for more information on using Contexts.
+//
+// NOTE(mhayden): Wait for 45 minutes for the snapshot to import.
+// Some AWS regions are fairly busy with image imports and they can
+// sometimes take a long time.
 func WaitUntilImportSnapshotTaskCompletedWithContext(c *ec2.EC2, ctx aws.Context, input *ec2.DescribeImportSnapshotTasksInput, opts ...request.WaiterOption) error {
 	w := request.Waiter{
 		Name:        "WaitUntilImportSnapshotTaskCompleted",
-		MaxAttempts: 80, // this can take up to 15 minutes, therefore set the timeout to 20 minutes
+		MaxAttempts: 180,
```
Comment (larskarlitski) [inline internal/upload/awsupload/awsupload.go:81]: I'm not sure we want a timeout here at all. (No blocker, can be done in a follow-up, especially if it needs more discussion.)

Comment (major) [reply internal/upload/awsupload/awsupload.go:81]: I don't know the AWS SDK well enough to know if we can completely remove the timeout. I'll go back and read the docs.

**Pull request merged**